nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,524 @@
+            description = ''
+              Domain part of the web interface URL (no web interface for this domain if null).
+              DNS record of type A (or AAAA) has to exist with this value.
+            '';
+          };
+          webLocation = mkOption {
+            type = str;
+            default = "/";
+            example = "/sympa";
+            description = "URL path part of the web interface.";
+          };
+          settings = mkOption {
+            type = attrsOf (either (either str int) bool);
+            default = {};
+            example = {
+              default_max_list_members = 3;
+            };
+            description = ''
+              The robot.conf configuration file as key value set.
+            '';
+          };
+        };
+
+        config.settings = mkIf (cfg.web.enable && config.webHost != null) {
+          wwsympa_url = mkDefault "https://${config.webHost}${strings.removeSuffix "/" config.webLocation}";
+        };
+      }));
+
+      description = ''
+        Email domains handled by this instance. There have
+        to be MX records for keys of this attribute set.
+      '';
+      example = literalExample ''
+        {
+          "lists.example.org" = {
+            webHost = "lists.example.org";
+            webLocation = "/";
+          };
+          "sympa.example.com" = {
+            webHost = "example.com";
+            webLocation = "/sympa";
+          };
+        };
+      '';
+    };
+
+    database = {
+      type = mkOption {
+        type = enum [ "SQLite" "PostgreSQL" "MySQL" ];
+        default = "SQLite";
+        example = "MySQL";
+        description = "Database engine to use.";
+      };
+
+      host = mkOption {
+        type = nullOr str;
+        default = "127.0.0.1";
+        description = "Database host address.";
+      };
+
+      port = mkOption {
+        type = nullOr port;
+        default = null;
+        description = "Database port.";
+      };
+
+      name = mkOption {
+        type = str;
+        default = if cfg.database.type == "SQLite" then "${dataDir}/sympa.sqlite" else "sympa";
+        defaultText = ''if database.type == "SQLite" then "${dataDir}/sympa.sqlite" else "sympa"'';
+        description = ''
+          Database name. When using SQLite this must be an absolute
+          path to the database file.
+        '';
+      };
+
+      user = mkOption {
+        type = nullOr str;
+        default = cfg.user;
+        defaultText = "user";
+        description = "Database user. The system user name is used as a default.";
+      };
+
+      password = mkOption {
+        type = str;
+        default = "";
+        description = ''
+          The password corresponding to <option>database.user</option>.
+          Warning: this is stored in cleartext in the Nix store!
+          Use <option>database.passwordFile</option> instead.
+        '';
+      };
+
+      passwordFile = mkOption {
+        type = nullOr path;
```

Comment:
Use `nullOr str` instead. This will prevent the file from ending up in the Nix store if people do `passwordFile = /run/keys/sympa-dbpassword`

Reply:
Hmm, good point! 
There are 14 more instances of `passwordFile` declared as `nullOr path` in current nixpkgs, should these be changed as well?

Reply:
Yes.

Reply:
Well actually, I seem to have spread misinformation, see https://github.com/NixOS/nixpkgs/pull/66346#issuecomment-520146519

`types.path` isn't in any way securer than `types.str`